### DISCUSSION TITLE
AG Latency Handling in Logic

### DISCUSSION THREAD

**Discussion by Pete3030** [apohl79/audiogridder#1048] [Help] [answered]
Hi, I have a problem with AG14 and latency handling within Logic. PDC is set to all. Logic shows me a latency of 32,6 ms for the AG Plugin. The AG Plugin is on an Audio channel with an External Input Plugin (which is receiving a channel of my HW mixing desk). When I duplicate this external instrument on a separate 2. Audio channel (with no AG plugin on it) and solo both channels, I can hear that they are not in sync. By deactivating (via on/off button) the AG plugin, both channels play in sync again. In the AG plugin both options for fixed size outbound buffers and allowing individual buffer sizes for plugin are checked, changing them in any combination does not fix the problem (I already tried that). To me it seems that Logic gets a wrong latency communicated by the AG plugin and therefore the PDC cannot work as it should. What surprises me that the situation can be healed by simply pressing the On/off button of the AG plugin. Any ideas what could be the problem here? Thank you in advance!

**Comment by apohl79** [comment, marked as answer]
Not sure if I understand the exact setup, but I think for live input there is no latency compensation. I tested this in logic with 2 audio tracks. I used an I/O plugin on both and routed external audio from my interface to it. (I hope thats what you do.) On one track I added AG. Once I set AG to use a buffer, I hear the delay. Deactivating the buffer or disabling AG brings both in sync. But I get the exact same behavior with any other plugin, that introduces a delay as well.

Also it behaves the same way as above if I route the same interface channel into the tracks input and enable input monitoring (no I/O plugin). And I confirmed that it is also exactly the same behavior on reaper.

But maybe thats not what you did?

**Comment by Pete3030** [comment]
Hi Andreas, sorry, you are totally right, I just tried with another high latency plugin and the effect is the same. Seems like I have to study Logics PDC a little more in detail. I somewhere read that you have to completely delete a plugin in order to get rid of it's induced latency in the channel, but it seems like the on/off button is already sufficient here. I will try working with AG with no Buffer at all, let's see how far my old Mac will be able to handle this in a mix setting..;) Sorry again, it was already late yesterday..

**Reply by apohl79** [reply]
Ok perfect. Glad there is no issue!
